packages/nuxt/src/runtime/plugins/sentry.server.ts

```diff
@@ -1,11 +1,48 @@
+import {
+  GLOBAL_OBJ,
+  consoleSandbox,
+  flush,
+  getClient,
+  getDefaultIsolationScope,
+  getIsolationScope,
+  logger,
+  vercelWaitUntil,
+  withIsolationScope,
+} from '@sentry/core';
 import * as Sentry from '@sentry/node';
 import { H3Error } from 'h3';
 import { defineNitroPlugin } from 'nitropack/runtime';
 import type { NuxtRenderHTMLContext } from 'nuxt/app';
 import { addSentryTracingMetaTags, extractErrorContext } from '../utils';
 
 export default defineNitroPlugin(nitroApp => {
-  nitroApp.hooks.hook('error', (error, errorContext) => {
+  nitroApp.h3App.handler = new Proxy(nitroApp.h3App.handler, {
+    async apply(handlerTarget, handlerThisArg, handlerArgs: Parameters<typeof nitroApp.h3App.handler>) {
+      // In environments where we cannot make use of OTel httpInstrumentation, e.g. when just importing the Sentry server config at
+      // the top level instead of `--import` or dynamic import like on Vercel, we still need to ensure requests are properly isolated
+      const isolationScope = getIsolationScope();
+      const newIsolationScope = isolationScope === getDefaultIsolationScope() ? isolationScope.clone() : isolationScope;
```

Comment:
maybe let's add a comment here that describes what/why we are doing here:

If otel instrumentation works, this will already be isolated at this point by the SentryHttpInstrumentation. We can identify this by comparing the current isolation scope to the default one. If that is the case, all good, we do not need to do anything. else, we fork the isolation scope here :)